[🔥AUDIT🔥] Update to latest octokit API

somewhatabstract opened the pull request:
🖍 _This is an audit!_ 🖍

## Summary:
The Octokit API has changed, so things are now off a rest endpoint. I must admit, I gave up on the flow types management and just installed the latest version from flow-typed which is next to useless. TypeScript upgrade would be nice right now.

I went in favour of getting things working instead. I can reinstate better typing at a later date if folks think I should.

Issue: FEI-4813

## Test plan:
`yarn test` (but these don't really test what we need to here so got to release it to main to find out).

Brianzchen commented:
Hello, was just stalking around flow-typed usage and came across this PR. I'm a flow-typed maintainer and I'm sorry the rest types weren't up to standards 😢 I added them myself and added just enough type definitions for my own needs.

My hope was that people who needed more than my initial use case would contribute back but I understand that it's sometimes a large ask considering how easy it is on the Typescript side.

I'll try look into further improving these types over the next couple weeks, hopefully that helps 🙏

somewhatabstract commented:
> Hello, was just stalking around flow-typed usage and came across this PR. I'm a flow-typed maintainer and I'm sorry the rest types weren't up to standards 😢 I added them myself and added just enough type definitions for my own needs.
> 
> My hope was that people who needed more than my initial use case would contribute back but I understand that it's sometimes a large ask considering how easy it is on the Typescript side.
> 
> I'll try look into further improving these types over the next couple weeks, hopefully that helps 🙏

Thanks for the work, @Brianzchen. I did not intend to be dismissive of your effort; I am sorry for doing that. My frustration was showing. 🙏🏻
